msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ManagedIdentityClient.java

```diff
@@ -65,4 +72,44 @@ private static AbstractManagedIdentitySource createManagedIdentitySource(MsalReq
                 return new IMDSManagedIdentitySource(msalRequest, serviceBundle);
         }
     }
+
+    static boolean validateAzureArcEnvironment(IEnvironmentVariables environmentVariables) {
+        if (!StringHelper.isNullOrBlank(environmentVariables.getEnvironmentVariable(Constants.IDENTITY_ENDPOINT)) &&
+                !StringHelper.isNullOrBlank(environmentVariables.getEnvironmentVariable(Constants.IMDS_ENDPOINT))) {
+            LOG.info("[Managed Identity] Azure Arc managed identity is available through environment variables.");
+            return true;
+        }
+
+        String osName = System.getProperty("os.name").toLowerCase();
+
+        if (osName.contains("windows")) {
```

Comment:
Can "windows" and "linux" be saved as constants in a constants file? Surely they are used elsewhere, or will be? Otherwise, if you implement my suggestion above, you could get the file path via:

```
// get the expected Windows or Linux file path of the himds executable
const fileDetectionPath: string =
    AZURE_ARC_FILE_DETECTION[process.platform as keyof FilePathMap];
```